packages/eui/src/components/steps/step.styles.ts

```diff
@@ -66,6 +68,20 @@ export const euiStepStyles = (euiThemeContext: UseEuiTheme) => {
         background-position: -${euiTheme.size.xs} ${euiTheme.size.l};
       }
     `,
+    // the xxs indicator circle is smaller than the text height, we realign the
+    // distances of the steps after manually horizontally aligning the step number
+    // to ensure the connection lines connect properly to the indicators
+    xxs: css`
+      &:not(:first-of-type) {
+        ${logicalCSS('margin-top', `calc(-1 * ${euiStep.numberXXSOffset})`)}
+      }
+
+      &:not(:last-of-type) {
+        /* Adjust the line to be centered on the smaller number */
+        background-position: -${euiTheme.size.s} calc(${euiTheme.size.base} +
+              ${euiStep.numberXXSOffset});
+      }
+    `,
```

Comment:
After having poked at this a bit more with the offset back in place, I'm honestly not a fan of having to shenanigan with the background and the margin offsets and so forth. Let's just go ahead and take the most straightforward option of reducing the line-height of horizontal xxs `titleSize`s to `1`/`16px` instead, which aligns them vertically and avoids this whole rigamarole.

```suggestion
    xxs: css`
      &:not(:last-of-type) {
        background-position: -${euiTheme.size.s} center;
    `,
```

Reply:
Hmm, this won't quite work though. Centering the background will result in the lines cutting the number indicators and not all have filled colors meaning the lines are visible 🙈 (see steps 4 and 7 in the image)

![Screenshot 2024-06-11 at 10 33 01](https://github.com/elastic/eui/assets/44670957/7811b1d6-b4cb-4792-81ce-59e0b7f2fc96)

We can remove the top offset in favor of the line-height but need to adjust the background-position-y anyway:

```suggestion
xxs: css`
      background-position: -${euiTheme.size.s} ${euiTheme.size.base};
`,
```

Reply:
Updated to use `line-height` in this [commit](https://github.com/elastic/eui/pull/7813/commits/24fb02444643f315d683bf69d9be2c9c27bae037).

Reply:
Gotcha, that makes total sense!! I wasn't super sure about the `center` personally, I was just kinda yolo guessing in devtools. Your snippet looks great!